enterprise/app/filter/filter.tsx

```diff
@@ -49,43 +68,88 @@ export default class FilterComponent extends React.Component<FilterProps, State>
     const startDate = (startDateParam ? moment(startDateParam) : moment().subtract(7, "days")).toDate();
     const endDate = (endDateParam ? moment(endDateParam) : moment()).toDate();
 
-    const isFiltering = Boolean(this.props.search.toString());
+    const canClearFilters = Boolean(this.props.search.toString());
+
+    const roleValue = this.props.search.get(ROLE_PARAM_NAME) || "";
 
     return (
-      <div className={`global-filter container ${isFiltering ? "is-filtering" : ""}`}>
-        {isFiltering ? (
-          <OutlinedButton
-            className="clear-filters-button"
-            title="Clear filters"
-            onClick={this.onClickClearFilters.bind(this)}>
-            <img src="/image/x.svg" alt="" />
-          </OutlinedButton>
-        ) : (
-          <div className="filter-icon-container">
-            <img src="/image/filter.svg" alt="" />
-          </div>
+      <div className={`global-filter ${canClearFilters ? "is-filtering" : ""}`}>
+        {canClearFilters && (
+          <FilledButton className="square" title="Clear filters" onClick={this.onClickClearFilters.bind(this)}>
+            <img src="/image/x-white.svg" alt="" />
+          </FilledButton>
         )}
-        <div className="date-picker-container">
-          <OutlinedButton className="date-picker-button" onClick={this.onOpenDatePicker.bind(this)}>
-            <img src="/image/calendar.svg" alt="" />
+        <div className="popup-wrapper">
+          <OutlinedButton className="filter-menu-button icon-text-button" onClick={this.onOpenFilterMenu.bind(this)}>
+            <img className="subtle-icon" src="/image/filter.svg" alt="" />
+            {roleValue === "CI" && <span className="role-badge CI">CI</span>}
+            {roleValue === "CI_RUNNER" && <span className="role-badge CI_RUNNER">Workflow</span>}
+            {!roleValue && <span className="filter-menu-button-label">Filter...</span>}
```

Comment:
Can we just make it the icon if no filter is selected? i.e. remove `Filter...`

Reply:
Done!